lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -69,24 +75,25 @@ def virtual_aggregate(name, relation, method_name = :sum, column = nil, options
         end
       end
 
+      def define_virtual_size_method(name, relation)
+        define_method(name) do
+          (has_attribute?(name) ? self[name] : send(relation).try(:size)) || 0
```

Comment:
this changed from `attribute_present` to `has_attribute`
Since a sql `count(*)` is never `null`, this will never be `nil`. So the `nil` check is pointless.

Reply:
This comment is covered in commit `virtual_total nil attribute does not exec query`